hub/config/src/main/java/uk/gov/ida/hub/config/data/ManagedEntityConfigRepository.java

```diff
@@ -36,27 +39,61 @@ public boolean has(String entityId){
     }
 
     public Optional<T> get(String entityId) {
-        RemoteConfigCollection remoteConfigCollection = s3ConfigSource.getRemoteConfig();
-        Optional<T> localConfig = localConfigRepository.getData(entityId);
-        return localConfig.map(local -> overrideWithRemote(local, remoteConfigCollection));
-    }
-
-    private T overrideWithRemote(T local, RemoteConfigCollection remoteConfigCollection){
-        if (local.isSelfService()){
-            return remoteConfigCollection.getRemoteComponent(local)
-                    .map(remote -> local.override(
-                            remote.getSignatureVerificationCertificates(),
-                            remote.getEncryptionCertificate(),
-                            CertificateOrigin.SELFSERVICE))
-                    .orElseGet(()-> {
-                        LOG.warn("Local config for '{}' expects there to be remote config but it could not be found", local.getEntityId());
-                        return local;
-                    });
-        }
-        return local;
+        return localConfigRepository.getData(entityId)
+                .map(this::overrideWithRemote);
     }
 
     public Stream<T> stream() {
         return localConfigRepository.stream();
     }
-}
\ No newline at end of file
+    
+
+    private T overrideWithRemote(T local) {
+        if (!local.isSelfService()) return local;
+        
+        return s3ConfigSource.getRemoteConfig()
+                .getRemoteComponent(local)
+                .map(remote -> getRemoteOverrides(local, remote))
+                .orElseGet(() -> logWarningAndReturnUnOverriddenConfig(local));
+    }
+
+    private T logWarningAndReturnUnOverriddenConfig(T local) {
+        LOG.warn("Local config for '{}' expects there to be remote config but it could not be found", local.getEntityId());
+        return local;
+    }
+
+    private T getRemoteOverrides(T local, RemoteComponentConfig remote) {
+        return local.override(
+                getSignatureVerificationCertificates(local, remote),
+                getEncryptionCertificate(local, remote),
+                CertificateOrigin.SELFSERVICE);
+    }
+
+    private List<String> getSignatureVerificationCertificates(T local, RemoteComponentConfig remote) {
+        try {
+            return remote.getSignatureVerificationCertificates();
+        }
+        catch (NullPointerException e){
+            LOG.warn("Remote config signing certificates missing for {}",
+                    Optional.ofNullable(local.getSignatureVerificationCertificates()
+                            .stream().map(Certificate::getSubject)
+                            .reduce("", String::join))
+                            .orElseGet(local::getEntityId));
```

Comment:
How does this work if the local config (which I presume to be the cert in the config file in `verify-hub-federation-config`) is missing?  The way I read this is that it falls back on the `v-h-f-c` cert if the remote one is missing.

Reply:
If it is missing then `.orElseGet(local::getEntityId));` it uses the `entity_id`